### ISSUE TITLE
Indented code blocks with special characters cause an error

### ISSUE THREAD

**Issue by pelson** [KyleKing/mdformat-mkdocs#31]
With ``mdformat-mkdocs``, having indented code with special characters results in a rendering problem:

```
Error: Could not format "example.md".

The formatted Markdown renders to different HTML than the input Markdown. This
is likely a bug in mdformat. Please create an issue report here, including the
input Markdown: https://github.com/executablebooks/mdformat/issues
```

The example:

```
# Title

1.  First step:

     ```bash
     echo \
      <container name>
     ```
```

I'm on Python 3.11, with:

```
$ pip list
Package         Version
--------------- -------
linkify-it-py   2.0.3
markdown-it-py  3.0.0
mdformat        0.7.17
mdformat_admon  2.0.3
mdformat-gfm    0.3.6
mdformat_mkdocs 2.1.0
mdformat_tables 0.4.1
mdit-py-plugins 0.4.1
mdurl           0.1.2
more-itertools  10.2.0
uc-micro-py     1.0.3
```


I also get a warning:

```
Warning: Plugin conflict. More than one plugin defined a renderer for "link" syntax.
```

I'm guessing this isn't related, but wish I knew how to remove the warning.

**Comment by pelson** [comment]
I should say, there is no problem formatting:

```
# Title

1.  First step:

     ```bash
     echo \
      "hello world"
     ```
```

**Comment by pelson** [comment]
> Hm, there is always an edge case with markdown 😅

Indeed :joy: 

FWIW, this is fine with vanilla mdformat.

**Comment by KyleKing** [comment]
This issue *should* be fixed with changes from #32 in v2.1.1!

**Comment by KyleKing** [comment]
FYI: the Warning for multiple definitions of 'href' should now be resolved with [v3.0.0](https://github.com/KyleKing/mdformat-mkdocs/releases/tag/v3.0.0)!

**Comment by pelson** [comment]
Verified with 3.0... it is looking awesome! Thanks for the great project!
